online_meetings/bluejeans_report.py

```diff
@@ -32,141 +37,267 @@
+                            })
+    return df
+
+# the main function to call for BlueJeans report
+def run_bluejeans_report() -> Sequence[Dict[str, Union[str, pd.Timestamp]]]:
+    logger.info("run_bluejeans_report")
+
+    total_df = pd.DataFrame()
+
+    # 1: get the access token
+    token = _1_get_bluejeans_api_token()
+    if not token:
+        logger.error("No API token. Exit for now. ")
+        return
+    
+    # use token for request header
     payload = {}
     headers = {
         'Accept': 'application/json',
         'Content-Type': 'application/json',
         'Authorization': f'Bearer {token}'
     }
-    r = requests.get(download_url, headers=headers)
 
-    # specifying the zip file name
-    zip_file_name = "./bj.zip"
+    # get the start and end time for report
+    from_date = dateparser.parse(ENV.get('BLUEJEANS_FROM_DATE', '2020-03-31')).date()
+    to_date = dateparser.parse(ENV.get('BLUEJEANS_TO_DATE', '2020-04-09')).date()
 
-    with open(zip_file_name, 'wb') as f:
-        f.write(r.content)
-    
-    # opening the zip file in READ mode
-    with ZipFile(zip_file_name, 'r') as zip_file:
-        # printing all the contents of the zip file
-        zip_file.printdir()
-    
-        # extracting all the files
-        print('Extracting all the files now...')
-        zip_file.extractall()
-        for csv_file in glob.glob("./meetings_*.csv"):
-            logger.info(csv_file)
-            df = pd.read_csv(csv_file)
-
-            # append to the total output
-            total_df = total_df.append(df)
-            logger.info(df)
-
-            # remove file
-            os.remove(csv_file)
-
-    # remove zip file
-    os.remove(zip_file_name)
-    print(f" {zip_file_name} file Removed!")
-    
+    # Only use the date as a paramter
+    # Loop until yesterday (this will go until now() -1)
+    for i in range((to_date - from_date).days):
+        param_date = from_date + timedelta(days=i)
+        start = datetime(param_date.year, param_date.month, param_date.day, 0, 0, 0)
+        end = datetime(param_date.year, param_date.month, param_date.day, 23, 59, 59)
+        logger.info("start")
+        logger.info(start.astimezone().isoformat())
+        logger.info("end")
+        logger.info(end.astimezone().isoformat())
+        
+        # post the report request job
+        report_url = ENV["BLUEJEANS_URL_REPORT"] + "?filter=\
+                    [{\"type\":\"date\",\"comparison\":\"lt\",\"value\":\"" + end.astimezone().isoformat() + "\",\"field\":\"end_time\"},\
+                        {\"type\":\"date\",\"comparison\":\"lt\",\"value\":\"" + end.astimezone().isoformat() + "\",\"field\":\"start_time\"},\
+                        {\"type\":\"date\",\"comparison\":\"gt\",\"value\":\"" + start.astimezone().isoformat() + "\",\"field\":\"start_time\"}\
+                    ]&fileName=meetings_30th-Mar-2020_30th-Mar-2020&userid=" + str(ENV.get("BLUEJEANS_USER_ID")) + "&app_name=command_center"
```

Comment:
This code generates a string with a lot of whitespace included.  Like…

```txt
?filter=                    [{"type":"date","comparison":"lt","value":"2020-03-01T23:59:59-05:00","field":"end_time"},                        {"type":"date","comparison":"lt","value":"2020-03-01T23:59:59-05:00","field":"start_time"},                        {"type":"date","comparison":"gt","value":"2020-03-01T00:00:00-05:00","field":"start_time"}                    ]&fileName=meetings_30th-Mar-2020_30th-Mar-2020&userid=None&app_name=command_center
```

This URL construction could be a lot easier to read if you changed the outer quotes from double `"` to single `'` (it will prevent the need for backslashes `\` in the string) and break it into pieces.  You can also eliminate the need for backslashes at line breaks after the `+` sign by wrapping the whole string in parentheses…

```suggestion
        report_url = (
            ENV['BLUEJEANS_URL_REPORT'] +
            '?filter=[' +
            '{"type":"date","comparison":"lt","value":"' +
            end.astimezone().isoformat() +
            '","field":"end_time"},' +
            '{"type":"date","comparison":"lt","value":"' +
            end.astimezone().isoformat() +
            '","field":"start_time"},' +
            '{"type":"date","comparison":"gt","value":"' +
            start.astimezone().isoformat() +
            '","field":"start_time"}' +
            ']&fileName=meetings_30th-Mar-2020_30th-Mar-2020&userid=' +
            str(ENV.get('BLUEJEANS_USER_ID')) +
            '&app_name=command_center'
        )
```

I was going to suggest using Python format strings (`f'…'`), but the curly braces in your strings caused problems with that.

In general, I suggest you avoid using `"` anywhere in Python programs where `'` can be used instead.  This is an example of why.

An even better solution that this would be to use Python dictionaries to hold all the URL parameters and use a URL construction module to build the URL for you.

Reply:
@zqian, I think you overlooked this comment.